### PR TITLE
feat: adds supporter badge

### LINKED ISSUE #2130: Add "supporter" badge to profiles
### Description
We want to introduce a `badge` for people supporting on [Patreon](https://www.patreon.com/one_army). (we recently changed our packages there so people can support specific projects) This is the start of intergrating memberships into the platform, for now the link between Patreon and the Platform will be done manually and it only shows a badge in the profile. But we can continue to build on this in the future

### Page related

Badge will be visable in : Member + Space Profiles `https://dev.onearmy.world/u/username`
Badge will be assigned in: User Profile Edit `https://dev.onearmy.world/u/username/edit` (new issue https://github.com/ONEARMY/community-platform/issues/2131)

### Mockup
supporter badge

### Build suggestion
We already have `badges` intergrated and use it to show `verified` users.  Have a look at that code to intergrate it

Icon.svg download here
[supporter.svg.zip](https://github.com/ONEARMY/community-platform/files/10833946/supporter.svg.zip)

### PR DESCRIPTION
**Pull request by thisislawatts**
PR Checklist

- [X] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

- docs: adds story showcasing all available icons
- feat: adds supporter badge to space and member profile

To review this locally you will need to set `badges.supporter` to true on your user document. 

## Git Issues

Closes #2130

## Screenshots/Videos

 Screenshot 2023-02-26 at 18 37 00

Screenshot 2023-02-26 at 18 36 29

### CODE REVIEW

**Comment by davehakkens** [comment]
looking good. Forgot to mention in the original Issue:
Would it be possible to link the "supporter" field to Patreon? https://www.patreon.com/one_army
Just like the location field (Netherlands) links to the url on the map

**Comment by thisislawatts** [comment]
@davehakkens updated with a through to Patreon

**Comment by AlfonsoGhislieri** [comment]
@davehakkens asked if I could give a hand reviewing, so I can have a look at this PR!

**Comment by thisislawatts** [comment]
Thanks @AlfonsoGhislieri, let me know if you've got any questions.

File: src/pages/User/content/MemberProfile.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ const MemberPicture = styled('figure')`
 `
 
 export const MemberProfile = ({ user }: IProps) => {
-  const userLinks = user?.links.filter(
+  const userLinks = (user?.links || []).filter(
```
Comment (AlfonsoGhislieri) [inline src/pages/User/content/MemberProfile.tsx:37]: Is it necessary to run `filter` if `user.links` has no values? Maybe could be worth using a conditional operator instead for this variable assignment, might help with readability.

**Review by AlfonsoGhislieri — COMMENTED** [review]
Looks good to me! Could be worth changing the `userLinks` assignment when `user.links` has no links.

File: src/pages/User/content/MemberProfile.tsx (hunk 1 above)
Comment (thisislawatts) [reply src/pages/User/content/MemberProfile.tsx]: Thanks @AlfonsoGhislieri, I have removed this line. I did some hacking on my local user object so it was in a slightly broken state where there was no links property. This component needs to be more defensive but I will save that for a separate PR.

**Review by davehakkens — APPROVED** [review]

**Pull request merged**